gpii/node_modules/gpii-db-operation/src/DbDataStoreUtils.js

```diff
@@ -74,6 +74,37 @@ gpii.dbOperation.dbDataStore.findRecord = function (dataSource, directModel, val
     return promiseTogo;
 };
 
+/**
+ * Similar to `gpii.dbOperation.dbDataStore.findRecord` but for CouchDB dataSources meant to return
+ * a list of records.
+ *
+ * Eventually this method, and the related CouchDB dataSources will take a standard set of options
+ * for paging and sorting. Because the dataProcessFunc can be operating on each record, ideally this
+ * should always return a reasonable amount of records you'd want to display say, on a page.
+ *
+ * @param {Component} dataSource - An instance of gpii.dbOperation.dbDataSource.
+ * @param {Object} directModel - The direct model expressing the "coordinates" of the model to be fetched.
+ * @param {Function} dataProcessFunc - The function to further process the retrieved record when the returned
+ * record is not empty.
+ * @return {Promise} A promise for the retrieved record.
+ */
+gpii.dbOperation.dbDataStore.findRecordList = function (dataSource, directModel, dataProcessFunc) {
```

Comment:
I don't see a case where `dataProcessFunc` is provided and used. This argument should be removed if it's not in use.

Reply:
For the `dataProcessFunc`, I added it to match the similar method `findRecord` because it seems useful for flexible processing.  From grepping the code base though, it doesn't look like anyone is exercising it when using `findRecord` either. 

The reason that I added the `findRecordList` in addition to `findRecord`, is that it is meant to always return a list, whereas `findRecord` only returns a single item. It's also slightly different when processing as well, because `findRecordList` can be empty as an acceptable value, whereas `findRecord` has special processing whenever something is empty to make it undefined.